afids_utils/afids.py

```diff
@@ -0,0 +1,59 @@
+"""Anatomical fiducial classes"""
+from __future__ import annotations
+
+import attr
+import numpy as np
+import polars as pl
+from numpy.typing import NDArray
+
+from afids_utils.exceptions import InvalidFiducialError
+
+
+@attr.define
+class AfidSet(dict):
```

Comment:
Why is this a dict subclass? Note that attrs has [`asdict`](https://www.attrs.org/en/stable/examples.html#converting-to-collections-types) if it's just for JSON conversion.

Reply:
The class itself is was meant to be a dict to split between the `metadata` and the `afids`. Without subclassing, the post init item assignment doesn't work (afaik). Open to alternatives though.

Reply:
I guess I'm not sure why we need that, but if it's useful we could add a third attrs class called AfidsMetadata with `slicer_version` and `coord_system` as attributes, then we won't need the post init

Reply:
I think the metadata is not immediately useful at the moment aside from dealing with the coordinate systems, but I think it ultimately will be down the line. So then are you thinking something along the lines of:

```python

class AfidSet:
    metadata: AfidsMetadata(slicer_version, coord_system, ...)
    afids: list[AfidPosition]
```

Reply:
That is fair - the first iteration of the implementation was how you had it. My thought on the change was to make the clear distinction between what is metadata (especially as additional fields are added) and what are the afids.

Reply:
Gotcha. My leaning would be to leave the metadata ungrouped if there's no immediate need to deal with it outside the context of an AfidSet, but ultimately up to you, I'm fine either way

Reply:
I'll leave it out for now, easier to handle - and also not a big change if we do want to group it down the line.